libcaf_net/caf/net/ip.hpp

```diff
@@ -30,6 +30,10 @@ namespace ip {
 /// Returns all IP addresses of to `host` (if any).
 std::vector<ip_address> resolve(string_view host);
 
+/// Returns the IP addresses for a local endpoint, which is either an address,
+/// an interface, or localhost.
```

Comment:
The function is meant to be used when creating local endpoints while resolve should be used for remote endpoints. That is why I would prefer an interface that mirrors the one of resolve.

Now we should answer the question if it would make sense for both functions to have  overloads for IP addresses and hosts.

* For `resolve` this would mean converting the ip address to a string because that is what `getaddrinfo` expects. 
* For `local_addresses` we could check first if the string is actually an IP address and dispatch that to the other overload. The result would be that both function call slightly different filters.

Having the overloads could save a back and forth conversion between IP address and string. Sounds like plan.